### PR TITLE
Add support for Basilisp file extension to Calva Spritz

### LINKED ISSUE #2629: Synchronize the file extensions for Calva and Calva Spritz
The Calva and Calva Spritz extensions have diverged a bit in what file types they are active on. This can create a rather weird experience. We should synchronize them.

See also:
* #2628

### PR DESCRIPTION
**Pull request by alendit**
## What has changed?

Calva supports many different runtimes, e.g. basilisp, via nrepl. However, Spritz doesn't recognize some of its extension. 

This change syncs the extensions between `package.json` and Spritz's `package.json` to enable notebook support for more file types. In particular, I add the following extensions clojure, joke, fiddle, ly, and lpy. 

Fixes #2629 

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Made sure there is an issue registered with a clear problem statement that this PR addresses, (created the issue if it was not present).
    - [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [x] Added to or updated docs in this branch, if appropriate
- [x] Tests
  - [x] Tested the particular change
  - [x] Figured if the change might have some side effects and tested those as well.
- [x] Formatted all JavaScript and TypeScript code that was changed. (use the [prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) or run `npm run prettier-format`)
- [x] Confirmed that there are no linter warnings or errors (use the [eslint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint), run `npm run eslint` before creating your PR, or run `npm run eslint-watch` to eslint as you go).

Ping @pez, @bpringe, @corasaurus-hex, @Cyrik

### CODE REVIEW

**Comment by PEZ** [comment]
Hello. Thanks for the PR! 🙏  I note that there are more discrepancies between the two manifests. I created an issue for it now. Maybe you can close that issue with this PR?

Also, please include the PR template again and go through the items in the checklist. (I've retargeted this PR at `dev` already, so that we don't accidentally merge it to `published`, which is known to have happened and it causes a lot of extra work for me.)

**Comment by PEZ** [comment]
Thanks for updating! I think `.fiddle` files are still lacking in the pattern.

Also, the changelog isn't updated yet. The release script needs that to know what to put in the release notes.

**Comment by alendit** [comment]
Hi @PEZ, yeah, somehow the codespaces didn't commit the CHANGELOG change.

I wasn't sure if `.fiddle` makes sense in the context of notebooks, but I can add it.

**Comment by alendit** [comment]
Turns out you need to explicitly push from the codespace. 

Should I add `.fiddle`?

**Comment by alendit** [comment]
Added fiddle, as well.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ Changes to Calva.
 
 ## [Unreleased]
 
+- [Sync the extensions between Calva an Calva Spritz](https://github.com/BetterThanTomorrow/calva/issues/2629)
```
Comment (PEZ) [inline CHANGELOG.md]:
```suggestion
- [Synchronize the file extensions for Calva and Calva Spritz](https://github.com/BetterThanTomorrow/calva/issues/2629)
```

**Comment by PEZ** [comment]
Awesome! I don't use notebooks a lot, but intuitively fiddle files and notebooks go together like a horse and carriage. =)

**Pull request merged**